nationalsCodev1.ino

```diff
@@ -1,85 +1,95 @@
-  //Encoder//
+/* Wiring Information */
 
+  /* Encoder */
     // Red - 5V
     // Black - GND
-    //Signal A in D2
-    const int encoder_a = 2; 
-    //Signal B in D3
-    const int encoder_b = 3; 
+    // Signal A in D2
+    const int ENCODER_A = 2; 
+    // Signal B in D3
+    const int ENCODER_B = 3; 
 
-  //Motor Shield//
+  /* Motor Shield */
 
-    // Motor PWM is connected to pin 5 (Check jumper).
-    const int pinPwm = 5;
-    // Motor DIR is connected to pin 4 (Check jumper).
-    const int pinDir = 4;
-////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Motor PWM - connected to pin 5 (Check jumper).
+    const int PIN_PWM = 5;
+    // Motor DIR - connected to pin 4 (Check jumper).
+    const int PIN_DIR = 4;
 
-//Setting initial values to zero//
-long encoder = 0;
-long REAL_DISTANCE = 0;
+// Set initial values
+void reset() {
+  long ENCODER_VALUE = 0;
+  long REAL_DISTANCE = 0; 
+}
 
-//Setting everything up//
+// Setup function
 void setup() {
-Serial.begin(9600); // Prepares the serial outputter //
-pinMode(encoder_a, INPUT_PULLUP); //Configures the encoder signal as an input//
-pinMode(encoder_b, INPUT_PULLUP); //Configures the encoder signal as an input//
 
-attachInterrupt(0, encoderPinChangeA, CHANGE); //Configures the encoder signal as an "interrupt"//
-attachInterrupt(1, encoderPinChangeB, CHANGE); //Configures the encoder signal as an "interrupt"//
+  // Call the reset
+  reset();
+
+  // Error handling
+  Serial.begin(9600);
 
-pinMode(pinPwm, OUTPUT); //Configures motor speed as output//
-pinMode(pinDir, OUTPUT); //Configures motor direction as output//
+  // Attach encoders
+  pinMode( ENCODER_A, INPUT_PULLUP );
+  pinMode( ENCODER_B, INPUT_PULLUP );
+
+  // Attach encoder interrupt
+  attachInterrupt(0, encoderPinChangeA, CHANGE );
+  attachInterrupt(1, encoderPinChangeB, CHANGE );
+
+  // Motor Speed
+  pinMode(PIN_PWM, OUTPUT);
+
+  // Motor Direction
+  pinMode(PIN_DIR, OUTPUT);
 }
 
-//Repeats over and over again//
+// Loop - FOREVER
 void loop() {
-//Printing data//
-  Serial.print("Encoder value"); //Prints label//
-  Serial.print(encoder); //Prints encoder value//
-  REAL_DISTANCE = abs(((encoder/600)*(3.14159*0.123825))) ///Conversion///
-  Serial.print("Real Distance value (m)"); //Prints label//
-  Serial.println(REAL_DISTANCE); //Prints distance value//
-//Controlling the motor based on the encoder value//
-  if (REAL_DISTANCE < change_speed_distance)
-    {
-      analogWrite(pinPwm, 200); //fast speed//
-      digitalWrite(pinDir, LOW);
-    }
-  else
-    {
-      analogWrite(pinPwm, 50); //slow speed//
-      digitalWrite(pinDir, LOW);
-    }
+
+  // Print encoder value
+  Serial.print("Encoder value: ");
+  Serial.println(ENCODER_VALUE);
+
+  // Calculate real distance
+  REAL_DISTANCE = abs( ( ( encoder / 600 ) * ( 3.14159 * 0.123825 ) ) )
```

Comment:
The encoder sends out 600 pulses per rotation. The wheels travel pi*d per rotation.